src/domain/community/organization/organization.service.ts

```diff
@@ -307,9 +308,28 @@ export class OrganizationService {
     return organization;
   }
 
-  async getOrganizations(): Promise<IOrganization[]> {
-    const organizations = await this.organizationRepository.find();
-    return organizations || [];
+  async getOrganizations(
+    limit?: number,
+    randomSelection = false
+  ): Promise<IOrganization[]> {
+    const organizations: IOrganization[] =
+      await this.organizationRepository.find();
+    if (!organizations) return [];
+    if (!limit || organizations.length < limit) return organizations;
+
+    // Need to restrict the set of users to return
+    if (randomSelection) {
+      const randomIndexes = generateRandomArraySelection(
+        limit,
```

Comment:
in order to get the code executed through here, you need to pass the `limit` param.

Reply:
see above